### PR TITLE
Documentation: added example to the function hasfield. (From #41782)

### PR DESCRIPTION
**Pull request by ViralBShah**
Try to land https://github.com/JuliaLang/julia/pull/41782 addressing some of the comments in there by @mcabbott.

### CODE REVIEW

**Comment by mcabbott** [comment]
This looks identical to https://github.com/JuliaLang/julia/pull/41782? I have completely forgotten but my comment https://github.com/JuliaLang/julia/pull/41782#issuecomment-892708253 was about whether `false` means "it doesn't have such a field" or just the opposite of "there is sure to be such a field".

**Comment by ViralBShah** [comment]
Yes, it is identical, but I figured I needed to be able to modify it and do some rebasing and stuff - and wasn't sure if I could do it on the user's fork. Hence created a branch here.

File: base/reflection.jl [hunk 1]
```diff
@@ -194,8 +194,22 @@ fieldnames(t::Type{<:Tuple}) = ntuple(identity, fieldcount(t))
...
+See also [`fieldnames`](@ref), [`fieldcount`](@ref), [`hasproperty`](@ref)
+
 !!! compat "Julia 1.2"
      This function requires at least Julia 1.2.
+
+# Examples
+```jldoctest
+julia> struct Foo
+            bar::Int
+       end
```
Comment (vtjnash) [inline base/reflection.jl:206]:
```suggestion
       end

```

File: base/reflection.jl [hunk 2]
```diff
@@ -194,8 +194,22 @@ fieldnames(t::Type{<:Tuple}) = ntuple(identity, fieldcount(t))
 
 Return a boolean indicating whether `T` has `name` as one of its own fields.
 
+See also [`fieldnames`](@ref), [`fieldcount`](@ref), [`hasproperty`](@ref)
```
Comment (vtjnash) [inline base/reflection.jl:197]:
```suggestion
See also [`fieldnames`](@ref), [`fieldcount`](@ref), [`hasproperty`](@ref).
```

**Pull request merged**